src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
I think we should move to the [ES6 classes](http://javascriptplayground.com/blog/2014/07/introduction-to-es6-classes-tutorial/) instead of our custom CoreObject/Class implementation. They'll be more accessible and interchangeable as ES6 classes get more use, and our current implementation doesn't do anything special. It would change `init` to `constructor`, so we'd need to figure out what larger external implications that will have.